### ISSUE TITLE
containDeep doesn't compare Dates inside of objects correctly

### ISSUE THREAD

**Issue by WillFrew** [shouldjs/should.js#39]
I'm using node v0.10.36 with the latest version of should (v4.6.2) and am seeing the opposite result as I'd expect from containDeep when comparing Dates inside of objects.

Example:
```javascript
var firstDec = [{date: (new Date('2014-12-01 00:00:00'))}];
var secondDec = [{date: (new Date('2014-12-02 00:00:00'))}];
firstDec.should.not.containDeep(secondDec); // Throws an AssertionError
firstDec.should.containDeep(secondDec); // No AssertionError
```

Output:
```
AssertionError: expected [
  { date: Monday, December 01, 2014 00:00:00.000 GMT-0000 }
] not to contain [
  { date: Tuesday, December 02, 2014 00:00:00.000 GMT-0000 }
]
```

**Comment by btd** [comment]
Yes, that is bug. Will fix soon.

**Comment by btd** [comment]
Fixed in 4.6.3

**Comment by WillFrew** [comment]
Wow, that was ridiculously fast - Thanks!!
